pq-crypto/kyber_r3/kyber512r3_fq_avx2.S

```diff
@@ -0,0 +1,126 @@
+#include "kyber512r3_cdecl_avx2.h"
+
+/*****.include "fq.inc"*****/
+/***************************/
+.macro red16 r,rs=0,x=12
+vpmulhw     %ymm1,%ymm\r,%ymm\x
+.if \rs
+vpmulhrsw   %ymm\rs,%ymm\x,%ymm\x
+.else
+vpsraw      $10,%ymm\x,%ymm\x
+.endif
+vpmullw     %ymm0,%ymm\x,%ymm\x
+vpsubw      %ymm\x,%ymm\r,%ymm\r
+.endm
+
+.macro csubq r,x=12
+vpsubw      %ymm0,%ymm\r,%ymm\r
+vpsraw      $15,%ymm\r,%ymm\x
+vpand       %ymm0,%ymm\x,%ymm\x
+vpaddw      %ymm\x,%ymm\r,%ymm\r
+.endm
+
+.macro caddq r,x=12
+vpsraw      $15,%ymm\r,%ymm\x
+vpand       %ymm0,%ymm\x,%ymm\x
+vpaddw      %ymm\x,%ymm\r,%ymm\r
+.endm
+
+.macro fqmulprecomp al,ah,b,x=12
+vpmullw     %ymm\al,%ymm\b,%ymm\x
+vpmulhw     %ymm\ah,%ymm\b,%ymm\b
+vpmulhw     %ymm0,%ymm\x,%ymm\x
+vpsubw      %ymm\x,%ymm\b,%ymm\b
+.endm
+/***************************/
+
+
+.text
+reduce128_avx:
+#load
+vmovdqa         (%rdi),%ymm2
+vmovdqa         32(%rdi),%ymm3
+vmovdqa         64(%rdi),%ymm4
+vmovdqa         96(%rdi),%ymm5
+vmovdqa         128(%rdi),%ymm6
+vmovdqa         160(%rdi),%ymm7
+vmovdqa         192(%rdi),%ymm8
+vmovdqa         224(%rdi),%ymm9
+
+red16           2
+red16           3
+red16           4
+red16           5
+red16           6
+red16           7
+red16           8
+red16           9
+
+#store
+vmovdqa         %ymm2,(%rdi)
+vmovdqa         %ymm3,32(%rdi)
+vmovdqa         %ymm4,64(%rdi)
+vmovdqa         %ymm5,96(%rdi)
+vmovdqa         %ymm6,128(%rdi)
+vmovdqa         %ymm7,160(%rdi)
+vmovdqa         %ymm8,192(%rdi)
+vmovdqa         %ymm9,224(%rdi)
+
+ret
+
+.global cdecl(reduce_avx2_asm)
+.global _cdecl(reduce_avx2_asm)
+cdecl(reduce_avx2_asm):
+_cdecl(reduce_avx2_asm):
```

Comment:
why export two function names?

Reply:
It is updated as it is implemented in Crystals Kyber repo.